### PR TITLE
Create init command

### LINKED ISSUE #8: Create init command
Like many tools, an initialize command (init) would be super useful for setting up a new project with colonize.  This would create a generic .colonize.yaml in the root of the project, a config ('env') directory, and boilerplate default files in the root config.

The boilerplate files should be controlled by a colonize "template" that would define what files should be created when initializing a new project.

### PR DESCRIPTION
**Pull request by jyore**
fixes #8 

This creates the new command `init`, which will start an interactive process to initialize the Colonize environment. After all options have been confirmed, it will create the specified Environments directory and the .colonize.yaml file.

I am creating the pull request to let others take a look before merged, but I think it will need an automated test before being merged. A way to automatically accept defaults without running the interactive process would also be useful, perhaps.

### CODE REVIEW

File: initialize/init.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+
+func Run(c *config.ColonizeConfig, l log.Logger) error {
+
+  scan := bufio.NewScanner(os.Stdin)
+  l.Log("Initializing a new Colonize project")
+
+  // environments_dir
+  l.Print("Enter 'environments_dir' [env]: ")
+  c.Environments_Dir = scan_with_default(scan, "env")
```
Comment (jyore) [inline initialize/init.go]: Should probably move these hardcoded defaults to a struct, which will also help with the "accept all defaults" flag

Comment (craigmonson) [reply initialize/init.go]: You could also create a map, and iterate over that to print and grab input.

Comment (jyore) [reply initialize/init.go]: Yep, working on something that's a little more along these lines. Have another flight tonight, so will hopefully have it mostly done letter

**Comment by jyore** [comment]
Refactored this a bit to use reflection for getting defaults & deep copies. This means that the init command will not need to be modified if fields are added/removed to/from the config file parameters. You would simply need to modify the struct and the default values variable for that struct.

I just need to build out some tests and that should wrap this up

**Pull request merged**